main_logic/core.py

```
@@ -560,9 +581,167 @@ async def handle_input_transcript(self, transcript: str):
+            and self._transient_response_instruction_blocked_session_id != session_id
+        ):
+            self._transient_response_instruction_blocked_session_id = None
+
+        base_instructions = self._transient_response_instruction_base or getattr(self.session, "instructions", "") or ""
+        if instruction == self._transient_response_instruction_active:
+            return
+
+        transient_instructions = base_instructions.rstrip() + "\n" + instruction
+        try:
+            await self.session.update_session({"instructions": transient_instructions})
+        except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)